ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py

```diff
@@ -43,11 +43,19 @@ def underlay_ethernet_interface(self, data: InterfaceDescriptionData) -> str:
             - overlay_routing_protocol
             - type
             - vrf
+            - wan_carrier
+            - wan_circuit_id
         """
-        desc = self.underlay_ethernet_interfaces(link_type=data.link_type, link_peer=data.peer, link_peer_interface=data.peer_interface)
+        desc = self.underlay_ethernet_interfaces(
```

Comment:
Instead of modifying the legacy function below, you can detect wan_carrier and/or wan_circuit_id here and just append it to the generated description from the other function. This is needed in case of this being sub-classed by a customer.

Reply:
Yea, got it.